### ISSUE TITLE
Motion control joy/pad intermittent issue.

### ISSUE THREAD

**Issue by KozGit** [KozGit/DOOM-3-BFG-VR#62]
Intermittent issue with motion controls where the joystick ( or pad in the case of the vive ) axes don't seem to be correctly reporting zero values or just miss input changes, which can result in the player spinning in place or moving constantly when no control is being pressed.

**Comment by CarlKenner** [comment]
SteamVR reports buttons as having the same state they did before whenever the controller loses tracking.

There's nothing we can do about it except check if we've lost tracking and set all the buttons to zero.

**Comment by KozGit** [comment]
I'll check to verify if this is the issue.  I don't recall any tracking issues when this happens, time to do some logging :)

**Comment by CarlKenner** [comment]
Also, I'm having an issue where sometimes I run it and the Oculus Touch buttons don't respond, so I have to exit and run it again.

**Comment by KozGit** [comment]
So the issue appears to with the vive/openvr.  I did some testing and polled all vr events and tracking changes, as well as all values directly reported by the controller after populating the controller state structure.  The hung state occurs with no errors, vr events or state changes reported - it just keeps duplicating the same value until the controller is touched again.  Polling the controller to see if the pad is being touch returns a positive result when this happens, even when you are not touching it at all.

The only other game I know off the top of my head that is directly built against the openvr sdk is the other doom 3 mod (Leylands), and I was able to duplicate the same issue there as well.

It's almost impossible to hold your finger so steady on the pad that it will return the same axis value for more than a few polls in a row.  I put a (optional) counter in place that checks for exact duplicate non zero repeated values, and defaults the axis val to 0 while the threshold is exceeded and the value continues to repeat.  I really don't like this kind of solution, but I like the player rotation being stuck on even less.  Hopefully we can figure out whats going on with polled values from the sdk and remove this hack at some point, but for now I'm closing this issue.
